### PR TITLE
[Datadog] Add custom tag names for metric filters

### PR DESCRIPTION
**Pull request by phalbert**
# Description

This PR introduces a feature that allows clients to specify custom tag names for hosts, in addition to the default universal Datadog tags (env and service). This enhancement provides flexibility for clients who have hosts tagged with different tags than the default ones used by Datadog.

### Description
The current system defaults to using the universal Datadog tags env and service for host identification and selection. However, some clients (e.g., IU) may have hosts tagged with different tags. To address this need, the following changes have been made:

- Modified the kind selector to allow the specification of custom tag names.
- Default behavior remains unchanged, using env and service tags if no custom tags are specified.
- Updated the `datadogSelector` configuration to include fields for specifying custom tags.

```yaml
kind: serviceMetric
selector:
  query: "true"
  datadogSelector:
    metric: "avg:system.disk.used"
    env: 
      tag: "environment"  # Custom tag specified by the user
      values: "*"
    service: 
      tag: "servicename"  # Custom tag specified by the user
      values: "*"
    timeframe: 5
```

### Additional Notes
- **Backward Compatibility**: The implementation ensures backward compatibility with existing configurations that use only the default env and service tags.
- **Documentation**: Updated documentation and examples have been provided to guide users on how to specify custom tags in their configurations.

## Type of change

Please leave one option from the following and delete the rest:

- [x]  Breaking change (fix or feature that would cause existing functionality to not work as expected)

<h4> All tests should be run against the port production environment(using a testing org). </h4>

### Core testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync finishes successfully
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Scheduled resync able to abort existing resync and start a new one
- [ ] Tested with at least 2 integrations from scratch
- [ ] Tested with Kafka and Polling event listeners

### Integration testing checklist

- [x] Integration able to create all default resources from scratch
- [x] Resync able to create entities
- [x] Resync able to update entities
- [x] Resync able to detect and delete entities
- [x] Resync finishes successfully
- [x] If new resource kind is added or updated in the integration, add example raw data, mapping and expected result to the `examples` folder in the integration directory. 
- [x] If resource kind is updated, run the integration with the example data and check if the expected result is achieved
- [x] If new resource kind is added or updated, validate that live-events for that resource are working as expected
- [x] Docs PR link [here](https://github.com/port-labs/port-docs/pull/1536)

### Preflight checklist

- [ ] Handled rate limiting
- [ ] Handled pagination
- [ ] Implemented the code in async
- [ ] Support Multi account

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

### CODE REVIEW

File: integrations/datadog/client.py [hunk 1]
```diff
@@ -347,29 +354,33 @@ def get_env_tags(self, tags: dict[str, Any]) -> list[str]:
...
         services: list[dict[str, Any]],
         time_window: int,
+        env_tag: str = "env",
+        service_tag: str = "service",
     ) -> AsyncGenerator[list[dict[str, Any]], None]:
         """Helper function to fetch metrics for a list of services and provided environments."""
         logger.info(
             f"Fetching metrics for {len(services)} services and {len(envs_to_fetch)} environments"
         )
+        logger.info(f"env_tag: {env_tag}, service_tag: {service_tag}")
```
Comment (Tankilevitch) [inline integrations/datadog/client.py:372]: redundant? maybe add it as part of the log above

File: integrations/datadog/examples/mappings.yaml
Comment (Tankilevitch) [inline integrations/datadog/examples/mappings.yaml:1]: please add raw data example in another file

File: integrations/datadog/client.py [hunk 2]
```diff
@@ -404,48 +415,65 @@ async def _fetch_metrics_for_services(
...
+            DatadogResourceConfig, event.resource_config
+        ).selector.datadog_selector
+
+        logger.info(f"Fetching metrics for params: {params}")
+
+        env_tag, env_value = self.get_metric_config(params.env)
+        service_tag, service_value = self.get_metric_config(params.service)
+
+        logger.info(f"env_tag: {env_tag}, env_value: {env_value}")
+        logger.info(f"service_tag: {service_tag}, service_value: {service_value}")
```
Comment (Tankilevitch) [inline integrations/datadog/client.py:441]: I think this should be inside the `main.py` and not in the client. as the client needs to be just get the variables and not be familiar with the event.resource_config.

**Review by Tankilevitch — CHANGES_REQUESTED** [review]

File: integrations/datadog/client.py [hunk 3]
```diff
@@ -407,45 +411,62 @@ async def _fetch_metrics_for_services(
...
+            service_tag (str): The tag name for service.
+            service_value (str): The service value to filter by, or "*" for all services.
+            timeframe (int): Time window in minutes for fetching metrics.
 
         Yields:
             AsyncGenerator[list[dict[str, Any]], None]: Each individual metric as it's fetched.
         """
+        logger.info(f"Fetching metrics for query: {metric_query}")
+        logger.info(f"env_tag: {env_tag}, env_value: {env_value}")
+        logger.info(f"service_tag: {service_tag}, service_value: {service_value}")
```
Comment (PeyGis) [inline integrations/datadog/client.py:436]: Can these logs be on the same line?

**Review by PeyGis — CHANGES_REQUESTED** [review]

File: integrations/datadog/examples/metric.response.json
Comment (Tankilevitch) [inline integrations/datadog/examples/metric.response.json:1]: also add result expected json file

File: integrations/datadog/main.py [hunk 4]
```diff
@@ -85,14 +85,18 @@ async def on_resync_services(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
 async def on_resync_service_metrics(kind: str) -> ASYNC_GENERATOR_RESYNC_TYPE:
     dd_client = init_client()
 
-    params = typing.cast(
+    # Extract parameters from event.resource_config
```
Comment (Tankilevitch) [inline integrations/datadog/main.py:88]: redundant comment

Comment (phalbert) [reply integrations/datadog/main.py]: removed now

File: integrations/datadog/examples/metric.response.json
Comment (phalbert) [reply integrations/datadog/examples/metric.response.json:1]: added now

**Review by Tankilevitch — APPROVED** [review]
LGTM

**Pull request merged**